src/NetSapiensSharp/Connector.cs

```diff
@@ -82,16 +96,35 @@ public Request<Tresponse> CreateRequest<Tresponse>(string action_path) where Tre
         public IRestResponse<Tresponse> Send<Tresponse>(Request<Tresponse> request)
             where Tresponse : class, new()
         {
+            if (IsAccessTokenExpired())
+            {
+                _SessionToken = null;
+            }
+            int myRetryCount = 1;
+            RequestRetry:
+
             Authenticate();
             request._Request.AddHeader("Authorization", "Bearer " + _SessionToken);
```

Comment:
Change this to `request._Request.AddOrUpdateParameter("Authorization", "Bearer " + _SessionToken, ParameterType.HttpHeader);` so that you don't have to explicitly remove the **Authorization** header.

From comment for `AddHeader`: _Shortcut to **AddParameter(name, value, HttpHeader)** overload_